### PR TITLE
Fixed program and social username creation in realistic search data g…

### LINKED ISSUE #879: Update realistic user generation command for programs & social username
Two updates are needed in the gen_realistic_search_data command:

1. Fake programs should be set to live=True
1. A social username should be created for each fake user (ie: we need to make sure that profiles.api.get_social_username returns a value for these users)

### PR DESCRIPTION
**Pull request by gsidebo**
#### What are the relevant tickets?

Closes #879 

#### What's this PR do?

_(copied from issue)_
Two updates are needed in the gen_realistic_search_data command:

1. Fake programs should be set to live=True
2. A social username should be created for each fake user (ie: we need to make sure that profiles.api.get_social_username returns a value for these users)

I also added an argument that can be used to assign the 'staff' role to a user of a given username

#### Where should the reviewer start?

Run `delete_realistic_search_data`, then `gen_realistic_search_data`

#### How should this be manually tested?

- Check the index to see that users have a `username` value in their profile
- Check that the programs created have live=True

### CODE REVIEW

**Comment by aliceriot** [comment]
the fake programs are both live, and the users all have usernames.

**Comment by aliceriot** [comment]
looking at code now

File: profiles/management/commands/gen_realistic_search_data.py [hunk 1]
```diff
@@ -248,10 +275,13 @@ def handle(self, *args, **options):
         if len(user_data_list) == existing_fake_user_count and len(program_data_list) == existing_fake_program_count:
             self.stdout.write("Realistic users and programs appear to exist already.")
         else:
-            new_program_count = deserialize_program_data_list(program_data_list)
+            new_programs = deserialize_program_data_list(program_data_list)
+            if options.get('staff_user', None):
```
Comment (aliceriot) [inline profiles/management/commands/gen_realistic_search_data.py]: I think the default 2nd argument for `.get` is already `None`, is there a python subtlety I'm missing?

**Comment by aliceriot** [comment]
I think I'm running into an issue where the `social_auth` records aren't all being deleted, does that seem like something that's possible? I edited some of the users to try to figure out why I couldn't see their profiles on `/users/${username}`, which might have mucked something up.

**Comment by pdpinch** [comment]
@gsidebo you didn't answer my questions on this issue (#879). I don't understand why this needed to be worked on right now?

**Comment by gsidebo** [comment]
@pdpinch i knocked this one out really quickly. sorry, didn't see those questions. i'll answer them on the issue

**Comment by gsidebo** [comment]
@aliceriot did you happen to edit User.username for those users you edited?

**Comment by aliceriot** [comment]
@gsidebo yeah I did for one user. I ran the delete command, and then when I tried to generate the fake users again I got an error from the one user I had edited. I deleted her and deleted all the fake users, and then ran it again.

**Comment by aliceriot** [comment]
nvm I don't think this is actually a problem with your code.

**Comment by aliceriot** [comment]
@gsidebo This looks good, I just had one question, back to you!

**Comment by aliceriot** [comment]
looks like one linting error, after that's fixed it's a 👍 from me!

**Pull request merged**